src/messages/mod.rs

```diff
@@ -625,6 +625,8 @@ pub enum MessageContent {
     UserMessagePart {
         /// The hash of this user message.
         hash: Digest256,
+        /// The unique message ID of this user message.
+        msg_id: MessageId,
```

Comment:
I think it's too big of a change for now. Having it duplicated in the Request/Response is only an extra 32 bytes per message, and we're already repeating the message ID up to 52 times per message which is ~1.6KB.